modules/images/webp-uploads/image-edit.php

```diff
@@ -387,9 +388,8 @@ function webp_uploads_get_next_full_size_key_from_backup( $attachment_id ) {
  */
 function webp_uploads_restore_image( $attachment_id, $data ) {
 	$backup_sources = get_post_meta( $attachment_id, '_wp_attachment_backup_sources', true );
-
 	if ( ! is_array( $backup_sources ) ) {
-		return $data;
+		$backup_sources = array();
```

Comment:
I have the same question as Felix and would like to explore that edge case.

I tried this edge case in which the `_wp_attachment_backup_sources` meta value is null or not an array. As per the below code snippet, it will initialize an empty array for `$backup_sources`.

	if ( ! is_array( $backup_sources ) ) {
		$backup_sources = array();
	}

After this snippet, we have another snippet that checks two things if `full-orig` key is not set in array `$backup_sources` or it's not an array.

	if ( ! isset( $backup_sources['full-orig'] ) || ! is_array( $backup_sources['full-orig'] ) ) {
		return $data;
	}

In this edge case the function return `$data` so as I understood it is worth initializing an empty array instead of returning the `$data` as pervious it does.

Please let me know if I'm getting anything wrong.